src/components/ActionButton/ToggleActionButton.ts

```diff
@@ -12,6 +12,46 @@ export interface IToggleActionButtonOptions {
 }
 
 export class ToggleActionButton extends Component {
+    /**
+     * Create the deactivated state for a given ToggleActionButton
+     * @param button {ToggleActionButton}
+     */
+    static generateDeactivatedStateInstance(button: ToggleActionButton): IStatefulActionButtonState {
+        return {
+            icon: button.options.activateIcon,
+            tooltip: button.options.activateTooltip,
+            click: () => button.onClick(),
+        };
+    }
+
+    /**
+     * Create the activated state for a given ToggleActionButton
+     * @param button {ToggleActionButton}
+     */
+    static generateActivatedStateInstance(button: ToggleActionButton): IStatefulActionButtonState {
+        const activate = button.options.activate?.bind(this);
+        const deactivate = button.options.activate?.bind(this);
+        return {
+            onStateEntry: function () {
+                this.element.classList.add(ToggleActionButton.ACTIVATED_CLASS_NAME);
```

Comment:
I know it's a draft but i would definitely put these two functions as private members somewhere else in the class

Reply:
On the second PR, you'll see that the State are actually their own full-fledged classes, which I think make this change unnecessary